libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -1492,6 +1493,31 @@ bool AP_InertialSensor::gyro_calibrated_ok_all() const
     return (get_gyro_count() > 0);
 }
 
+// Prearm check to verify that we have sane sched loop rates set based on Gyro backend rates
+bool AP_InertialSensor::pre_arm_check_gyro_backend_rate_hz(char* fail_msg, uint16_t fail_msg_len) const
+{
+#if AP_SCHEDULER_ENABLED
+    for (uint8_t i=0; i<get_gyro_count(); i++) {
+        if (!_use(i) || _backends[i] == nullptr) {
+            continue;
+        }
+        // if gyro backend rate hz is 0, that means we are polling sensor at the sample rate or higher
+        if (_backends[i]->get_gyro_backend_rate_hz() == 0) {
+            if (_gyro_raw_sample_rates[i] < _loop_rate) {
```

Comment:
Don't you have to cater for oversampling here? See get_gyro_rate_hz()

Reply:
Didn't we agree that the rate needs to be at least 2x the loop rate?

Reply:
> Don't you have to cater for oversampling here? See get_gyro_rate_hz()

No, oversampling is only set by drivers that use FIFO, and hence will be setting backend_rate_hz.



> Didn't we agree that the rate needs to be at least 2x the loop rate?

Done.